addons/website/static/src/snippets/s_embed_code/000.js

```diff
@@ -0,0 +1,41 @@
+/** @odoo-module **/
+
+import publicWidget from "web.public.widget";
+import { _t } from "@web/core/l10n/translation";
+
+const EmbedCodeWidget = publicWidget.Widget.extend({
+    selector: ".s_embed_code",
+    disabledInEditableMode: false,
+
+    /**
+     * @override
+     */
+    async start() {
+        this.embedCodeEl = this.el.querySelector(".s_embed_code_embedded");
+        if (this.editableMode && this.embedCodeEl.offsetHeight === 0) {
+            this._addPlaceholder();
+        }
+        return await this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        this._super(...arguments);
+        const embedContent = this.embedCodeEl.dataset.embedContent;
+        if (!this.editableMode && embedContent && embedContent.match(/<script[^<]*?>/)) {
+            this.embedCodeEl.innerHTML = embedContent.replace(/<script[^<]*?>.*?<\/script>/gs, "");
+        }
+    },
+    _addPlaceholder() {
```

Comment:
I don't see the value in creating a function just for this, code would be easier to read with a good comment and this code directly in the `start`.